marketplace/commercelayer-products/src/index.js

```diff
@@ -0,0 +1,162 @@
+import CLayerAuth from '@commercelayer/js-auth';
+import CLayer from '@commercelayer/js-sdk';
+
+import { setup, renderSkuPicker } from 'shared-sku-app';
+
+import logo from './logo.svg';
+import descriptor from '../extension.json';
+import { dataTransformer, previewDataTransformer } from './dataTransformer';
+
+const PER_PAGE = 20;
+
+function makeCTA(fieldType) {
+  return fieldType === 'Array' ? 'Select products' : 'Select a product';
+}
+
+/**
+ * The Commerce Layer client is currently used to fetch only the previews
+ * for the selected products.
+ *
+ * To fetch all SKUs and to search through them we perform directly an HTTP
+ * request to the corresponding endpoint.
+ *
+ * @see fetchSKUs for a more detailed explanation.
+ */
+async function makeCommerceLayerClient({
+  parameters: {
+    installation: { apiEndpoint, clientId, clientSecret }
+  }
+}) {
+  const auth = await CLayerAuth.integration({
+    clientId,
+    clientSecret,
+    endpoint: apiEndpoint
+  });
+
+  CLayer.init({
+    accessToken: auth.accessToken,
+    host: apiEndpoint.replace(/(https?:\/\/)/g, '')
+  });
+}
+
+/**
+ * This function is needed to make the pagination of Commerce Layer work with the
+ * shared-sku-app library.
+ *
+ * When fetching the SKUs via the Commerce Layer JS SDK the metadata object which
+ * includes the total count of records needed by the shared-sku-picker paginator
+ * is missing. But it is there when fetching the SKUs via a plain HTTP req.
+ */
+async function fetchSKUs({ clientId, clientSecret, apiEndpoint }, search, pagination) {
+  const auth = await CLayerAuth.integration({
+    clientId: clientId,
+    clientSecret: clientSecret,
+    endpoint: apiEndpoint
+  });
+
+  const URL = `${apiEndpoint}/api/skus?page[size]=${PER_PAGE}&page[number]=${pagination.offset /
+    PER_PAGE +
+    1}${search.length ? `&filter[q][name_or_code_cont]=${search}` : ''}`;
+
+  const res = await fetch(URL, {
+    headers: {
+      Accept: 'application/vnd.api+json',
+      Authorization: `Bearer ${auth.accessToken}`
+    },
+    method: 'GET'
+  });
+
+  const result = await res.json();
+
+  return result;
+}
+
+/**
+ * Fetches the product previews for the products selected by the user
+ */
+const fetchProductPreviews = async function fetchProductPreviews(skus, config) {
+  if (!skus.length) {
+    return [];
+  }
+  await makeCommerceLayerClient({ parameters: { installation: config } });
```

Comment:
Since api installations can provide whatever they like, we should validate installation parameters before use. You also already have a method for that :)

Reply:
Done

Reply:
Why we don't pass `config` directly? I don't think `makeCommerceLayerClient` needs to know about either parameters or installation parameters.